coderd/database/queries/templates.sql

```diff
@@ -8,13 +8,39 @@ WHERE
 LIMIT
 	1;
 
--- name: GetTemplatesByIDs :many
+-- name: GetTemplatesWithFilter :many
 SELECT
 	*
 FROM
 	templates
 WHERE
-	id = ANY(@ids :: uuid [ ]);
+	-- Optionally include deleted templates
+	templates.deleted = @deleted
+	-- Filter by organization_id
+	AND CASE
+		WHEN @organization_id :: uuid != '00000000-00000000-00000000-00000000' THEN
+			organization_id = @organization_id
+		ELSE true
+	END
+	-- Filter by name, matching on substring
+	AND CASE
+		WHEN @name :: text != '' THEN
+			LOWER(name) LIKE '%' || LOWER(@name) || '%'
+		ELSE true
+	END
```

Comment:
Do we need this? We only do substring matching on name as a special case, we probably don't need it for other query fields thought.

Reply:
I can drop it for now. I will keep the other field as `exact_name` though

Reply:
When we add these search bars on other pages, we might want it back.